pkg/route/registry/route/strategy.go

```diff
@@ -43,14 +43,15 @@ func (routeStrategy) NamespaceScoped() bool {
 func (s routeStrategy) PrepareForCreate(obj runtime.Object) {
 	route := obj.(*api.Route)
 	route.Status = api.RouteStatus{}
+	// TODO: this does not belong here, and should be removed
+	shard, err := s.RouteAllocator.AllocateRouterShard(route)
```

Comment:
allocation only happens if no route.spec.host is set. if it is set, that should be treated as the fully qualified hostname (e.g. "www.foo.com")